spec/models/profile_spec.rb

```diff
@@ -1,5 +1,24 @@
 require 'rails_helper'
 
 RSpec.describe Profile, :type => :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+  it { should have_valid(:user_id).when('1', '100') }
+  it { should_not have_valid(:user_id).when(nil, '', 'asdf', 'dog') }
+
+  it { should have_valid(:blog_url).when('http://www.google.com', 'https://wwww.google.com') }
+  it { should_not have_valid(:blog_url).when(nil, '', 'asdf', 'dog', 'www.google.com') }
+
+  it { should have_valid(:facebook).when('http://www.google.com', 'https://wwww.google.com') }
+  it { should_not have_valid(:facebook).when(nil, '', 'asdf', 'dog', 'www.google.com') }
+
+  it { should have_valid(:twitter).when('http://www.google.com', 'https://wwww.google.com') }
+  it { should_not have_valid(:twitter).when(nil, '', 'asdf', 'dog', 'www.google.com') }
```

Comment:
Line is too long. [87/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.